src/main/webapp/script.js

```diff
@@ -71,22 +72,66 @@ function passQuery() {
     });
 }
 
-function displayAmChartsMap(data, description, location, isCountyQuery) {
-  am4core.useTheme(am4themes_animated);
-  const chart = am4core.create('map', am4maps.MapChart);
-  chart.height = 550;
-  chart.zoomControl = new am4maps.ZoomControl();
-  // only allow zooming with buttons
-  chart.mouseWheelBehavior = 'none';
-
-  // Create map instance
+function getGeoData(location, isCountyQuery) {
   if (isCountyQuery) {
-    chart.geodata = location ===
+    return location ===
         '06' ? am4geodata_region_usa_caLow : am4geodata_region_usa_njLow;
   } else {
-    chart.geodata = am4geodata_usaLow;
+    return am4geodata_usaLow;
   }
+}
+
+
+function displayVisualization(censusDataArray, description,
+  location, isCountyQuery) {
+  const geoData = getGeoData(location, isCountyQuery);
+  if (isCountyQuery) {
```

Comment:
This can be much simpler if the underlying functions like SetStyle can handle both cases.

Reply:
fixed to one function setStyle(isCountyQuery);